[receiver/mongodbreceiver] Integration test fails on TestMongodbIntegration

### Component(s)

receiver/mongodb

### What happened?

## Description
Integration test fails on `TestMongodbIntegration`

## Steps to Reproduce

## Expected Result
Passed the test
## Actual Result
```shell
--- FAIL: TestMongodbIntegration (0.00s)
    --- FAIL: TestMongodbIntegration/5_0 (50.27s)
        integration_test.go:181: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/integration_test.go:181
            	Error:      	Received unexpected error:
            	            	number of resources doesn't match expected: 10, actual: 0
            	Test:       	TestMongodbIntegration/5_0
    --- FAIL: TestMongodbIntegration/4_2 (50.54s)
        integration_test.go:181: 
            	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/mongodbreceiver/integration_test.go:181
            	Error:      	Received unexpected error:
            	            	the following errors occurred:
            	            	 -  resource "map[database:config]": scope "otelcol/mongodbreceiver": number of metrics doesn't match expected: 10, actual: 9
            	            	 -  resource "map[database:local]": scope "otelcol/mongodbreceiver": number of metrics doesn't match expected: 10, actual: 9
            	            	 -  resource "map[database:testdb]": scope "otelcol/mongodbreceiver": number of metrics doesn't match expected: 10, actual: 9
            	            	 -  resource "map[database:admin]": scope "otelcol/mongodbreceiver": number of metrics doesn't match expected: 10, actual: 9
            	Test:       	TestMongodbIntegration/4_2
FAIL
```



### Collector version

v0.75.0

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

Workflow: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4719638095/jobs/8370717385?pr=20767

--- a/receiver/mongodbreceiver/integration_test.go
+++ b/receiver/mongodbreceiver/integration_test.go
@@ -145,17 +145,20 @@ func (tt testCase) run(t *testing.T) {
 	require.NoError(t, err, "failed creating metrics receiver")
 
 	require.NoError(t, rcvr.Start(context.Background(), componenttest.NewNopHost()))
-
-	// Wait for multiple collections, in case the first represents partially started system
-	require.Eventuallyf(t, func() bool {
-		return len(consumer.AllMetrics()) > 1
-	}, 2*time.Minute, 1*time.Second, "failed to receive more than 0 metrics")
-	require.NoError(t, rcvr.Shutdown(context.Background()))
-	actualMetrics := consumer.AllMetrics()[1]
+	defer func() {
+		require.NoError(t, rcvr.Shutdown(context.Background()))
+	}()
 
 	expectedFile := filepath.Join("testdata", "integration", fmt.Sprintf("expected.%s.yaml", tt.name))
 	expectedMetrics, err := golden.ReadMetrics(expectedFile)
 	require.NoError(t, err)
+
+	// Wait for multiple collections, in case the first represents partially started system
+	require.Eventuallyf(t, func() bool {
+		return len(consumer.AllMetrics()) > 0 && consumer.AllMetrics()[len(consumer.AllMetrics())-1].MetricCount() == expectedMetrics.MetricCount()
+	}, 2*time.Minute, 1*time.Second, "failed to receive all metric data points")
+
+	actualMetrics := consumer.AllMetrics()[len(consumer.AllMetrics())-1]
 
 	require.NoError(t, pmetrictest.CompareMetrics(expectedMetrics, actualMetrics, pmetrictest.IgnoreMetricValues(),
 		pmetrictest.IgnoreMetricDataPointsOrder(), pmetrictest.IgnoreStartTimestamp(), pmetrictest.IgnoreTimestamp()))

